src/handlers/motions/motionStaked/helpers.ts

```diff
@@ -0,0 +1,183 @@
+import { BigNumber } from 'ethers';
+import { MotionQuery, MotionStakes, UserStakes } from '~types';
+import { getMotionSide } from '../helpers';
+
+export const getRequiredStake = (
+  skillRep: BigNumber,
+  totalStakeFraction: BigNumber,
+): BigNumber =>
+  skillRep.mul(totalStakeFraction).div(BigNumber.from(10).pow(18));
+
+export const getMotionStakes = (
+  requiredStake: BigNumber,
+  [totalNayStakesRaw, totalYayStakesRaw]: [BigNumber, BigNumber],
+): MotionStakes => {
+  const totalYayStakesPercentage = getStakePercentage(
+    totalYayStakesRaw,
+    requiredStake,
+  );
+
+  const totalNayStakesPercentage = getStakePercentage(
+    totalNayStakesRaw,
+    requiredStake,
+  );
+
+  const motionStakes: MotionStakes = {
+    raw: {
+      yay: totalYayStakesRaw.toString(),
+      nay: totalNayStakesRaw.toString(),
+    },
+    percentage: {
+      yay: totalYayStakesPercentage.toString(),
+      nay: totalNayStakesPercentage.toString(),
+    },
+  };
+
+  return motionStakes;
+};
+
+export const getRemainingStakes = (
+  requiredStake: BigNumber,
+  [totalNayStakesRaw, totalYayStakesRaw]: [BigNumber, BigNumber],
+): [string, string] => {
+  const remainingYayStake = requiredStake.sub(totalYayStakesRaw).toString();
+  const remainingNayStake = requiredStake.sub(totalNayStakesRaw).toString();
+  return [remainingNayStake, remainingYayStake];
+};
+
+export const getStakePercentage = (
+  stake: BigNumber,
+  requiredStake: BigNumber,
+): BigNumber => stake.mul(100).div(requiredStake);
+
+/**
+ * Given staking data, format and return new UserStakes object
+ */
+const getNewUserStakes = (
+  staker: string,
+  vote: BigNumber,
+  amount: BigNumber,
+  requiredStake: BigNumber,
+): UserStakes => {
+  const invertedVote = BigNumber.from(1).sub(vote);
+  const stakedSide = getMotionSide(vote);
+  const unstakedSide = getMotionSide(invertedVote);
+  return {
+    address: staker,
+    stakes: {
+      raw: {
+        [stakedSide]: amount.toString(),
+        [unstakedSide]: '0',
+      },
+
+      percentage: {
+        [stakedSide]: getStakePercentage(amount, requiredStake).toString(),
+        [unstakedSide]: '0',
+      },
+    }, // TS doesn't like the computed keys, but we know they're correct
+  } as unknown as UserStakes;
+};
+
+/**
+ * Given staking data, add stakes to existing stakes and return new, updated UserStakes object
+ */
+const getUpdatedUserStakes = (
```

Comment:
These two functions `getNewUserStakes` and `getUpdatedUserStakes` feel to me like really they're doing the work a good `class` should do. But we don't seem to write classes generally and I'm happy to leave as is if no objections.

Reply:
We do write them on occasions.

I personally don't think these two should be a class, and they can operate as-is, but if you think differently, and have a good argument for it, I'm good if you re-write them as a Class.